File: ios/RNContainer/ReactNativeViewController.m
#import "ReactNativeViewController.h"

#if __has_include(<React/RCTRootView.h>)
#import <React/RCTRootView.h>
#elif __has_include("RCTRootView.h")
#import "RCTRootView.h"
#else
#import "React/RCTRootView.h"   // Required when used as a Pod in a Swift project
#endif

#import "ReactNativeBridge.h"

@implementation ReactNativeViewController

@synthesize moduleName = _moduleName;
@synthesize initialProperties = _initialProperties;

-(instancetype)initWithModuleName:(NSString *)moduleName {
    return [self initWithModuleName:moduleName andInitialProperties:nil];
}

-(instancetype)initWithModuleName:(NSString *)moduleName
             andInitialProperties:(NSDictionary* _Nullable)initialProperties {
    self = [super init];
    _moduleName = moduleName;
    _initialProperties = initialProperties;
    
    return self;
}

-(void)viewDidLoad {
    if (_moduleName) {
      self.view = [[ReactNativeBridge shared]rootViewWithName:_moduleName properties:_initialProperties];
    }
}

@end

File: ios/RNContainer/ReactNativeViewController.swift
import React

public class ReactNativeViewController: UIViewController {
    private let moduleName: String
    private let initialProperties: [AnyHashable: Any]?

    public override func loadView() {
        guard let bridge = ReactNativeBridge.shared.bridge else {
            fatalError("You need to start React Native in order to use ReactNativeViewController, make sure to run ReactNativeBridge.shared.start() before instantiating it.")
        }
        view = RCTRootView(bridge: bridge, moduleName: moduleName, initialProperties: initialProperties)
    }
    
    public init(moduleName: String, initialProperties: [AnyHashable: Any]? = nil) {
        self.moduleName = moduleName
        self.initialProperties = initialProperties
        super.init(nibName: nil, bundle: nil)
    }
    
    @available(*, unavailable)
    required init?(coder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
